src/test/article.spec.js

```diff
@@ -143,3 +144,82 @@ describe('PATCH api/v1/articles/bookmark', () => {
     expect(newNumberOfBookmarks.count).to.equal(0);
   });
 });
+
+describe('GET api/v1/articles', () => {
+  beforeEach(async () => {
+    await db.Article.destroy({ cascade: true, truncate: true });
+    await db.User.destroy({ cascade: true, truncate: true });
+  });
+
+  it('should get a list of all articles', async () => {
+    const firstUser = await createUser();
+    const secondUser = await createUser();
+    const theArticle = await createArticle(
+      await generateArticle({ authorId: secondUser.id })
+    );
+
+    const numberOfArticles = await BaseRepository.findAndCountAll(db.Article);
+    const token = helper.jwtSigner(firstUser);
+    expect(numberOfArticles.count).to.equal(1);
+
+    const res = await server()
+      .get(`${ARTICLES_API}`)
+      .set('token', token);
+    expect(res.status).to.equal(200);
+    expect(res.body.data).to.be.an('array');
+    expect(res.body.data).to.have.lengthOf(1);
+    expect(res.body.data[0].id).to.equal(theArticle.id);
+    expect(res.body.data[0].authorId).to.equal(secondUser.id);
+    expect(res.body.data[0].title).to.equal(theArticle.title);
+    expect(res.body.data[0].body).to.equal(theArticle.body);
+    expect(res.body.data[0].image).to.equal(theArticle.image);
+  });
+
+  it('should list articles with pagaination', async () => {
+    const firstUser = await createUser();
+    const secondUser = await createUser();
+    const theArticle = await createArticle(
+      await generateArticle({ authorId: secondUser.id })
+    );
+
+    const numberOfArticles = await BaseRepository.findAndCountAll(db.Article);
+    const token = helper.jwtSigner(firstUser);
+    expect(numberOfArticles.count).to.equal(1);
+    const page = 1;
+    const limit = 1;
+    const paginate = new Pagination(page, limit);
+    const metadata = paginate.getPageMetadata(1, '/url');
+
+    expect(metadata.prev).to.equal(null);
+    expect(metadata.currentPage).to.equal(1);
+    expect(metadata.next).to.equal(null);
+    expect(metadata.totalPages).to.equal(1);
+    expect(metadata.totalItems).to.equal(1);
+
+    const res = await server()
+      .get(`${ARTICLES_API}`)
```

Comment:
Make the request look like this: *.get(`${ARTICLES_API}?limit=${limit}page=${page}`);* because this is the `PUBLIC API`.

Then assert that the metadata section of the response matches what is expected.